mountpoint-s3-client/src/object_client.rs

```diff
@@ -9,6 +9,11 @@ use time::OffsetDateTime;
 /// object and the bytes starting at that offset.
 pub type GetBodyPart = (u64, Box<[u8]>);
 
+#[derive(Debug, Clone)]
+pub struct ETag {
+    pub etag: String,
+}
```

Comment:
I'm a little hesitant to let the inner `etag` be `pub`. This means that anyone can depend on the internals of this structure.

Instead, I'd recommend something like this:

```suggestion
#[derive(Debug, Clone)]
pub struct ETag {
   etag: String,
}

impl ETag {
    pub fn as_str(&self) -> &str {
        &self.etag
    }
    
    pub fn from_str(value: &str) -> Self {
        ETag {
            etag: value
        }
    }
}
```

I purposely avoided recommending an implementation of `From<&str>` (which allows `"my-etag".into()`) from James' earlier comment about trying to avoid accidentally using the wrong value for etag. I don't particularly mind either way.